udata/core/spatial/search.py

```diff
@@ -43,15 +46,18 @@ class Meta:
                               payloads=True)
 
     @classmethod
-    def compute_weight(cls, population):
-        """Weight must be in the interval [0..2147483647]"""
-        if 0 <= population <= 2147483647:
-            return population
-        else:
-            if population < 0:  # country/eh population is -99.
-                return 0
-            else:  # World population is 6772425850.
-                return 2147483647
+    def compute_weight(cls, zone):
+        '''
+        Give a weight to the zone according its administrative level first
+        and then its population.
+        '''
+        # Each level give give a step
+        level = min(admin_levels.get(zone.level, ADMIN_LEVEL_MAX), 1)
+        level_weight = (ADMIN_LEVEL_MAX / level) * 10 * PONDERATION_STEP
```

Comment:
Because, of the use later: for the mathemtical explnation, th `* 10` is here to dilate the `(ADMIN_LEVEL_MAX / level)` which is `1 < f(level) < inf`. The factor is only here to ensure steps are not overlapping too much. If I touch the `PONDERATION_STEP`, the overlapping doesn't change, it is dilated too.